lib/StackAnalysis/StackAnalysis.cpp

```diff
@@ -441,6 +1975,244 @@ bool StackAnalysis::runOnModule(Module &M) {
               getName(Function.Entry) << (Function.Force ? " (forced)" : ""));
   }
 
+  // Initialize hook markers for subsequent ABI analysis on function calls
+  auto *FTy = llvm::FunctionType::get(llvm::Type::getVoidTy(M.getContext()),
+                                      { MetaAddress::getStruct(&M),
+                                        MetaAddress::getStruct(&M) },
+                                      false);
+  auto *PreHookMarker = Function::Create(FTy,
+                                         llvm::GlobalValue::ExternalLinkage,
+                                         "precall_hook",
+                                         M);
+  PreHookMarker->addFnAttr(llvm::Attribute::ReadOnly);
+  PreHookMarker->addFnAttr(llvm::Attribute::NoUnwind);
+
+  auto *PostHookMarker = Function::Create(FTy,
+                                          llvm::GlobalValue::ExternalLinkage,
+                                          "postcall_hook",
+                                          M);
+  PostHookMarker->addFnAttr(llvm::Attribute::ReadOnly);
+  PostHookMarker->addFnAttr(llvm::Attribute::NoUnwind);
+
+  auto *RetHookMarker = Function::Create(FTy,
+                                         llvm::GlobalValue::ExternalLinkage,
+                                         "retcall_hook",
+                                         M);
+  RetHookMarker->addFnAttr(llvm::Attribute::ReadOnly);
+  RetHookMarker->addFnAttr(llvm::Attribute::NoUnwind);
+
+  // Initialize marker to adjust `unexpectedpc` basic block for fake functions
+  FTy = llvm::FunctionType::get(llvm::Type::getVoidTy(M.getContext()), false);
+  auto
+    *UnexpectedPCMarker = Function::Create(FTy,
+                                           llvm::GlobalValue::ExternalLinkage,
+                                           "unexpectedpc_hook",
+                                           M);
+  UnexpectedPCMarker->addFnAttr(llvm::Attribute::ReadOnly);
+  UnexpectedPCMarker->addFnAttr(llvm::Attribute::NoUnwind);
+
+  // Queue to be populated with the CFEP
+  llvm::SmallVector<BasicBlock *, 8> Worklist;
+  llvm::SmallVector<BasicBlockNode *, 4> EntryBlocks;
+  UniquedQueue<BasicBlockNode *> CFEPQueue;
+  SmallCallGraph CG{};
+
+  using BasicBlockToNodeMapTy = llvm::DenseMap<BasicBlock *, BasicBlockNode *>;
+  BasicBlockToNodeMapTy BasicBlockNodeMap;
+
+  // Create over-approximated call graph
+  for (CFEP &Function : Functions) {
+    BasicBlockNode Node{ Function.Entry };
+    auto *GraphNode = CG.addNode(Node);
+    BasicBlockNodeMap[Function.Entry] = GraphNode;
+  }
+
+  for (CFEP &Function : Functions) {
+    llvm::SmallSet<BasicBlock *, 8> Visited;
+    auto *StartNode = BasicBlockNodeMap[Function.Entry];
```

Comment:
Yeah, no strong opinion here, but `assert`s are kind of defensive tools for future changes to the code.
If a certain piece of code assumes that *there must be* an entry in that map, if things above change in the future it will be beneficial to get an assertion.